src/scripts/generate-readme.ts

```diff
@@ -0,0 +1,106 @@
+import fs from 'node:fs'
+
+interface GetUsageSectionOptions<TNullable = false> {
+  isNullable?: TNullable
+}
+
+const getUsageSection = <TNullable = false>(
+  readmePath: string,
+  options: GetUsageSectionOptions<TNullable> = {},
+): TNullable extends true ? string | null : string => {
+  const { isNullable = false } = options
+
+  try {
+    return readUsageSection(readmePath)
+  } catch (error) {
+    if (isNullable === true) {
+      return null as any
```

Comment:
`any` is required to make the generic type-narrowing work. Open to suggestions here.